internal/config/server/config.go

```diff
@@ -137,3 +139,16 @@ func setViperStructDefaults(v *viper.Viper, prefix string, s any) {
 		v.SetDefault(valueName, defaultValue)
 	}
 }
+
+func overrideViperStructDefaults(v *viper.Viper, prefix string, newDefaults string) {
+	overrides := map[string]any{}
+	if err := json.Unmarshal([]byte(newDefaults), &overrides); err != nil {
+		panic(fmt.Sprintf("Failed to parse overrides in %q: %v", prefix, err))
+	}
```

Comment:
So this doesn't allow zeroing the config i.e. `default:"{}"`, is this by design? 
Following the current approach zeroing can be achieved by something like:
```go
Aggregator AggregatorConfig `mapstructure:"aggregator" default:"{\"lock_interval\":0}"`
```

Reply:
Yes, that is by design.  In most cases empty defaults are actually useless (as noticed when the `events.sql.connection` defaults for port and TLS were zeroed).  Making this zeroing explicit reduces the effort to produce a _valid_ configuration leveraging defaults.  (At a slightly higher cost of possible misconfiguration by omitting a value.)